pkg/tc/lock/lock_manager.go

```diff
@@ -30,7 +31,17 @@ func (locker *LockManager) AcquireLock(branchSession *apis.BranchSession) bool {
 		return true
 	}
 
-	return locker.manager.AcquireLock(locks)
+	applicationData := branchSession.ApplicationData
+	if applicationData != nil {
```

Comment:
TCC mode also uses `branchSession.ApplicationData`, so it's better to check `branchSession.Type`
```suggestion
	if applicationData != nil && branchSession.Type == apis.AT {
```